src/request.rs

```diff
@@ -108,6 +108,9 @@ macro_rules! lsp_request {
     ("textDocument/typeDefinition") => {
         $crate::request::GotoTypeDefinition
     };
+    ("textDocument/selectionRange") => {
+        $crate::request::SelectionRange
```

Comment:
Should be `SelectionRangeRequest`

Reply:
yeah...

I hate how some of these are `XRequest` and some are just `X`, but this inconsistent naming comes from the protocol itself. 

I wonder if "maintaining consistency with inconsistent protocol" is worth it...

Reply:
Whatever makes it easier I think. I'd recommend using the macros to refer to them in most cases anyway since that reflects their actual RPC name. So the name of the underlying type shouldn't matter most of the time.